### DISCUSSION TITLE
very low fps

### DISCUSSION THREAD

**Discussion by JELLYGAMESSE** [pokepetter/ursina#303] [Q&A]
Hey! in my game on ursina there is little fps, there is nowhere to cut the game, I achieved 50-60 fps, but if you start to move and even more so to shoot, it will be 30 fps,
and if another player connects to online, the game will specifically lag.

How to increase FPS? I read that the geometry of the collisions is very slow, but I did not find how to replace it in the official manual.
Please help to increase the FPS in the game.

**Comment by pokepetter** [comment]
You must have added something to make it slow. So what did you add?
Try to use as few Entites and colliders as possible.

**Reply by JELLYGAMESSE** [reply]
my imports:

import os
import sys
import socket
import threading
import ursina
from ursina import *
from network import Network

there are not so many objects in the game:
![Screenshot_28](https://user-images.githubusercontent.com/63513844/139706391-345e883a-7564-4a5b-b96b-95cfdf70a17e.png)

as code:

# карта "паркур_1"
            Platform_5(ursina.Vec3(0, -2, 0))
           
            
            Platform_5(ursina.Vec3(-8, -2, 0))

            
            
            Platform_1(ursina.Vec3(6, 2, 8))
            Platform_2(ursina.Vec3(-14, -1, 8))


            Platform_4(ursina.Vec3(6, -1, -8))
            Platform_4(ursina.Vec3(-14, -1, -8))

            Platform_3(ursina.Vec3(-7, 2, -16))
            Platform_3(ursina.Vec3(-7, 2, -20))
            Platform_3(ursina.Vec3(-7, 2, -24))


            Platform_3(ursina.Vec3(-7, 2, 16))
            Platform_3(ursina.Vec3(-7, 2, 20))
            Platform_3(ursina.Vec3(-7, 2, 24))
            Wall_info(ursina.Vec3(-5, 3, 16))


            Barrier(ursina.Vec3(0, -10, 30))
            Barrier(ursina.Vec3(0, -10, -30))

            Barrier_90G(ursina.Vec3(30, -10, 0))
            Barrier_90G(ursina.Vec3(-30, -10, 0))
            

            getUp(ursina.Vec3(-5, -5, 16))

**Reply by pokepetter** [reply]
Yeah, that doesn't look like it should be slow. What's the code for the platforms? Which part of the code is making it slow? Are you using box colliders?  Have you made sure it's not creating new Entities in update? Why does it lag more when moving as opposed to standing still? What's the difference?

**Reply by JELLYGAMESSE** [reply]
ч
this is how the object looks like a type:
class Wall(ursina.Entity):
    def __init__(self, position):
        super().__init__(
            position=position,
            scale=ursina.Vec3(2, 2, 2),
            model="cube",
            texture=os.path.join("assets", "wall.png"),
            origin_y=-0.5
        )
        #self.texture.filtering = None
        self.collider = ursina.BoxCollider(self, size=ursina.Vec3(1, 2, 1))
        
        
        

and then to add an object to the scene, I write:

Wall(ursina.Vec3(6, y, 0))


it is possible that the multiplayer is working, a large number of packages may slow down the game, but decreasing the number of packages / data will increase the ping.



object "Platform" 
class Platform_6(ursina.Entity):
    def __init__(self, position):
        super().__init__(
            position=position,
            scale=ursina.Vec3(4, 2, 4),
            model="cube",
            texture=os.path.join("assets", "floor.png"),
            origin_y=-0.5
        )
        #self.texture.filtering = None
        self.collider = ursina.BoxCollider(self, size=ursina.Vec3(1, 2, 1))

**Reply by JELLYGAMESSE** [reply]
sorry for wasting your time :)
I'm only 15 years old and I don't have enough experience to solve this alone

**Comment by 3110vaibhav2005** [comment]
I am also suffering from same problem of fps. In my game while i am hitting with entity the fps decrease to 10-5 fps and it lag very much...
Please help mee
